packages/web-api/src/converters/scan-run-error-converter.ts

```diff
@@ -1,11 +1,25 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 import { injectable } from 'inversify';
-import { ScanRunErrorCode, ScanRunErrorCodes } from 'service-library';
+import { isNil, isString } from 'lodash';
+import { scanErrorNameToErrorMap, ScanRunErrorCode, ScanRunErrorCodes } from 'service-library';
+import { ScanError } from 'storage-documents';
 
 @injectable()
 export class ScanRunErrorConverter {
-    public getScanRunErrorCode(scanError: string): ScanRunErrorCode {
+    public getScanRunErrorCode(scanError: string | ScanError): ScanRunErrorCode {
+        if (isNil(scanError)) {
+            return ScanRunErrorCodes.internalError;
+        }
+
+        if (isString(scanError)) {
+            return this.resolveLegacyErrorType(scanError);
+        }
+
+        return scanErrorNameToErrorMap[scanError.errorType];
+    }
+
+    private resolveLegacyErrorType(scanError: string): ScanRunErrorCode {
```

Comment:
May be call it parseErrorString()?

Reply:
i want to remove this type after a month. so calling it legacy for now